src/main/java/duke/storage/TaskStorage.java

```diff
@@ -0,0 +1,103 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.exception.DukeOperationException;
+import duke.exception.DukeParseException;
+import duke.parser.StorageParser;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Class that loads and converts the <code>TaskList</code> in Duke with a text file.
+ */
+public class TaskStorage {
+    private static final String DEFAULT_FILENAME = "taskstorage.txt";
+    private static final String DEFAULT_FILEPATH = "src/main/java/duke/storage/";
+
+    private final File file;
+    private final StorageParser storageParser;
+
+    private TaskStorage(File file) {
+        this.file = file;
+        this.storageParser = new StorageParser();
+    }
+
+    /**
+     * Creates a <code>TaskStorage</code>.
+     * The path to the file used is pre determined by default.
+     * If the path directory does not exist, the file will then be saved into the root of the directory.
+     *
+     * @return a preconfigured <code>TaskStorage</code>.
+     */
+    public static TaskStorage createTaskStorage() {
+        File f = new File(DEFAULT_FILEPATH);
+        File actualFile = f.exists()
+                ? new File(DEFAULT_FILEPATH + DEFAULT_FILENAME)
+                : new File(DEFAULT_FILENAME);
+        return new TaskStorage(actualFile);
+    }
```

Comment:
It will be better to just standardize the save location. So if the directory does not exist, create it and just save the file to the same directory. This can help reduce confusion on the save location for both user and programmer.

Reply:
Yup! I think Yan Cheng brought up a very good point. Allowing users to input their distinct file paths may improve the flexibility of your storage class, especially in the case where you may be required to store different types of data in different data files. Perhaps it may be possible to integrate this change by taking in a string function parameter for your createTaskStorage method where the string represents the file path.